src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -2681,407 +1707,102 @@ void ObjectSynchronizer::audit_and_print_stats(bool on_exit) {
   guarantee(error_cnt == 0, "ERROR: found monitor list errors: error_cnt=%d", error_cnt);
 }
 
-// Check a free monitor entry; log any errors.
-void ObjectSynchronizer::chk_free_entry(JavaThread* jt, ObjectMonitor* n,
-                                        outputStream * out, int *error_cnt_p) {
-  stringStream ss;
-  if (n->is_busy()) {
-    if (jt != NULL) {
-      out->print_cr("ERROR: jt=" INTPTR_FORMAT ", monitor=" INTPTR_FORMAT
-                    ": free per-thread monitor must not be busy: %s", p2i(jt),
-                    p2i(n), n->is_busy_to_string(&ss));
-    } else {
-      out->print_cr("ERROR: monitor=" INTPTR_FORMAT ": free global monitor "
-                    "must not be busy: %s", p2i(n), n->is_busy_to_string(&ss));
-    }
-    *error_cnt_p = *error_cnt_p + 1;
-  }
-  if (n->header().value() != 0) {
-    if (jt != NULL) {
-      out->print_cr("ERROR: jt=" INTPTR_FORMAT ", monitor=" INTPTR_FORMAT
-                    ": free per-thread monitor must have NULL _header "
-                    "field: _header=" INTPTR_FORMAT, p2i(jt), p2i(n),
-                    n->header().value());
-      *error_cnt_p = *error_cnt_p + 1;
-    }
-  }
-  if (n->object_peek() != NULL) {
-    if (jt != NULL) {
-      out->print_cr("ERROR: jt=" INTPTR_FORMAT ", monitor=" INTPTR_FORMAT
-                    ": free per-thread monitor must have NULL _object "
-                    "field: _object=" INTPTR_FORMAT, p2i(jt), p2i(n),
-                    p2i(n->object_peek()));
-    } else {
-      out->print_cr("ERROR: monitor=" INTPTR_FORMAT ": free global monitor "
-                    "must have NULL _object field: _object=" INTPTR_FORMAT,
-                    p2i(n), p2i(n->object_peek()));
-    }
-    *error_cnt_p = *error_cnt_p + 1;
-  }
-}
-
-// Lock the next ObjectMonitor for traversal and unlock the current
-// ObjectMonitor. Returns the next ObjectMonitor if there is one.
-// Otherwise returns NULL (after unlocking the current ObjectMonitor).
-// This function is used by the various list walker functions to
-// safely walk a list without allowing an ObjectMonitor to be moved
-// to another list in the middle of a walk.
-static ObjectMonitor* lock_next_for_traversal(ObjectMonitor* cur) {
-  assert(is_locked(cur), "cur=" INTPTR_FORMAT " must be locked", p2i(cur));
-  ObjectMonitor* next = unmarked_next(cur);
-  if (next == NULL) {  // Reached the end of the list.
-    om_unlock(cur);
-    return NULL;
-  }
-  om_lock(next);   // Lock next before unlocking current to keep
-  om_unlock(cur);  // from being by-passed by another thread.
-  return next;
-}
+// Check the in_use_list; log the results of the checks.
+void ObjectSynchronizer::chk_in_use_list(outputStream* out, int *error_cnt_p) {
+  size_t l_in_use_count = _in_use_list.count();
```

Comment:
Style nit: what is this `l_` prefix? Is that a one or a  small L? why do we want want/need it? (Applies elsewhere too)

Reply:
The "l_" prefix is used for a local copy of a value where we want to
make sure that we use a consistent value for the check and for the
resulting audit/logging message.

This is not a new thing with this changeset and that style was used
in previous versions of the ObjectMonitor audit/logging code.